add a new machine named M5C for ankermake

slice-engine-dev opened the pull request:
add a new machine name M5C for ankermake .The latest upload includes modifications to the anker.ini configuration file and the addition of five new files, which comprise two SVG files, two PNG files, and one STL document.

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
 # All presets starting with asterisk, for example *common*, are intermediate and they will
 # not make it into the user interface.
+[printer_model:M5C]
+    name = AnkerMake M5C
```
Please remove the leading spaces. This is inconsistent with other configurations and makes it difficult to review as it is hard to see what values actually changed.

slice-engine-dev replied:
OK

just-trey replied:
I still see the leading spaces in this area.

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+    inherits = *0.30mm*
+    compatible_printers_condition = printer_notes=~/.*(MACHINE_M5_).*/ and nozzle_diameter[0]==0.4
+
+[print:Precision - M5C (0.4 mm nozzle) @ANKER]
```
This should inherit common and only change the values needed for this mode

slice-engine-dev replied:
it means that it inherits *0.30mm*，and *0.30mm* inherits common

just-trey replied:
you are correct. I misread this one.

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+
+[print:Precision - M5C (0.4 mm nozzle) @ANKER]
+    compatible_printers_condition = printer_notes=~/.*(MACHINE_M5C_).*/ and printer_notes=~/.*(P_PRECISION).*/ and nozzle_diameter[0]==0.4
+
```
extra space not needed

slice-engine-dev replied:
OK

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+    fuzzy_skin_thickness = 0.3
+    gap_fill_enabled = 1
+    gap_fill_speed = 75
+
```
extra space not needed

slice-engine-dev replied:
thanks, OK

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+    slowdown_external_perimeters = 0
+
+[print:Normal - M5C (0.4 mm nozzle) @ANKER]
+    inherits = *param 0.4 P_NORMAL 0.20*
```
The configuration `0.4 P_NORMAL 0.20` does exist. Are you meaning to inherit *common*?

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+
+[print:Normal - M5C (0.4 mm nozzle) @ANKER]
+    inherits = *param 0.4 P_NORMAL 0.20*
+    notes = PRINT_COMPATIBLE\n P_NORMAL\n F_PLA__Basic\n
```
Are you sure you want to limit the use of this configuration to PLA only?

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+
+    #add by august<
+    #print---august
+    avoid_crossing_curled_overhangs = 0
```
This entire section should inherit a valid base configuration item and only change the values needed for this mode.

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+[print:Fast - M5C (0.4 mm nozzle) @ANKER]
+    inherits = *param 0.4 P_FAST 0.25*
+
+    #add by august<
```
This entire section should inherit a valid base configuration item and only change the values needed for this mode.

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+    slowdown_external_perimeters = 0
+
+[print:Fast - M5C (0.4 mm nozzle) @ANKER]
+    inherits = *param 0.4 P_FAST 0.25*
```
this configuration being inherited does not exist

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -186,87 +847,87 @@ compatible_printers_condition = printer_model=~/(M5).*/ and nozzle_diameter[0]==
 # So having some leeway to get good bed adhesion is not a luxury for many users
 
 [filament:*common*]
```
the extra prefix spaces make it hard to see what has changed

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -186,87 +847,87 @@ compatible_printers_condition = printer_model=~/(M5).*/ and nozzle_diameter[0]==
-min_print_speed = 30
-slowdown_below_layer_time = 8
-compatible_printers_condition = printer_notes=~/.*PRINTER_VENDOR_ANKERMAKE.*/
+    cooling = 1
```
I would set cooling to 0 and overwrite where it needs to be 1, or just set to 0 for ABS

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -289,78 +950,140 @@ filament_vendor = Generic
+                machine_limits_usage = time_estimate_only
+            ## Maximum feedrates
+                 machine_max_feedrate_e = 100
+                 machine_max_feedrate_x = 500,500
```
why do all these parameters have 2 values? Theses machines do not have stealth mode

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -289,78 +950,140 @@ filament_vendor = Generic
-retract_before_wipe = 0%
-printer_notes = Don not remove the following keywords! These keywords are used in the "compatible printer" condition of the print and filament profiles to link the particular print and filament profiles to this printer profile.\nPRINTER_VENDOR_ANKERMAKE\nPRINTER_MODEL_M5
+    inherits = *common*
+    bed_shape = 0x0,235-0,235x235,0x235
```
you are redefining values with the same value

just-trey reviewed (changes requested):
The inheritance of parameter values needs another look. There is a lot of duplication that is not needed throughout the file.

slice-engine-dev commented:
I have made the requested changes. Could you please review them again?@just-trey

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -166,16 +178,265 @@ top_solid_layers = 4
-compatible_printers_condition = printer_model=~/(M5).*/ and nozzle_diameter[0]==0.4
+compatible_printers_condition = printer_notes=~/.*(MACHINE_M5_).*/ and nozzle_diameter[0]==0.4
+
+[print:*common-M5C*]
```
the propet way to add this would be to `inherits = *common*` then overwrite the values you want to change.

slice-engine-dev replied:
The 3D printer M5C inherits its printing parameters from *common-M5C*, while the M5 inherits its printing parameters from *common*.

just-trey commented on resources/profiles/Anker.ini:
```diff
@@ -26,156 +26,817 @@ default_materials = Generic PLA+ @ANKER; Generic PLA @ANKER; Generic PET @ANKER;
+    inherits = *0.30mm*
+    compatible_printers_condition = printer_notes=~/.*(MACHINE_M5_).*/ and nozzle_diameter[0]==0.4
+
+[print:Precision - M5C (0.4 mm nozzle) @ANKER]
```
this looks good to me, be sure to "comment and close" when you see this message.

just-trey reviewed (changes requested):
Just a couple of changes for you to look at.

slice-engine-dev commented:
Thanks for the review!

just-trey commented:
> Thanks for the review!

Sorry about the close issue. There is a way to mark uems resolved but I do not see it in my review end

augustchi commented:
Apologies for the inadvertent closure of the issue/PR. I have reopened it . Thank you for your patience and guidance.Thank you for your suggestion on the proper way to structure the inheritance. I have followed your advice and implemented it accordingly. The 3D printer M5C now inherits its printing parameters from common-M5C, and the M5 inherits its printing parameters from common. I have made sure to override the specific values that needed to be changed for each printer model. I have removed the leading spaces as requested . Please let me know if there are any further adjustments required

slice-engine-dev commented:
Apologies for the inadvertent closure of the issue/PR. I have reopened it . Thank you for your patience and guidance.Thank you for your suggestion on the proper way to structure the inheritance. I have followed your advice and implemented it accordingly. The 3D printer M5C now inherits its printing parameters from common-M5C, and the M5 inherits its printing parameters from common. I have made sure to override the specific values that needed to be changed for each printer model. I have removed the leading spaces as requested . Please let me know if there are any further adjustments required

rtyr commented:
@ankermake Are you OK with this PR?

slice-engine-dev commented:
Yes, I'm OK with this PR

slice-engine-dev commented:
@rtyr

rtyr commented:
I was asking @ankermake as AnkerMake bundle was initially and officially submitted by them.

slice-engine-dev commented:
Hi, @rtyr. Apologies for any confusion. I am a member of the AnkerMake team and have been authorized by my company to make changes and submit code on behalf of the team. I have reviewed the PR and confirm that I am okay with the proposed changes. Please let me know if there is any additional information you need from me or if there are any further steps I should take.

rtyr commented:
I increased the version and released the update. Now I noticed, you have incorrect value for top solid infill extrusion width (0.2) in M5C print profiles.

rtyr commented:
I will fix it and release another update. It is not possible to slice anything with current 0.2mm value.

rtyr commented:
Fixed to 0.4mm (same as M5).

slice-engine-dev commented:
Sure, 0.4 is fine. I had changed it in the Ankermake software before, but I forgot to update it here because the submission was too early. Sorry for the inconvenience.@rtyr
